tsdb/head.go

```diff
@@ -805,6 +805,26 @@ func (h *Head) Init(minValidTime int64) error {
 	if err != nil && err != record.ErrNotFound {
 		return errors.Wrap(err, "find last checkpoint")
 	}
+
+	// Find the last segment.
+	// The purpose of this scope is to avoid wal.Segments err override wal.LastCheckpoint err.
+	var endAt int
+	{
+		var err error
+		_, endAt, err = wal.Segments(h.wal.Dir())
+		if err != nil {
+			return errors.Wrap(err, "finding WAL segments")
+		}
+	}
+
+	// Not likely endAt will smaller then startFrom,
+	// as the head cut a new wal segment file before creating a new checkpoint.
+	if endAt < startFrom {
+		endAt = startFrom
+	}
```

Comment:
If someone manually deletes some WAL files (which is not so rare), this case is possible and it changes the behaviour below and can cause errors. We should not change this here, and instead handle the min being greater than max in `startWALReplayStatus`.

Reply:
Very nice consideration 👍